javalib/src/main/scala/java/util/regex/Pattern.scala

```diff
@@ -82,11 +85,33 @@ object Pattern {
   final val CANON_EQ = 0x80
   final val UNICODE_CHARACTER_CLASS = 0x100
 
-  def compile(regex: String, flags: Int): Pattern =
-    new Pattern(regex, flags)
+  def compile(regex: String, flags: Int): Pattern = {
+    val (jsPattern, flags1) = {
+      if ((flags & LITERAL) != 0) {
+        (quote(regex), flags)
+      } else {
+        trySplitHack(regex, flags) orElse
+        tryFlagHack(regex, flags) getOrElse
+        (regex, flags)
+      }
+    }
+
+    val jsFlags = {
+      var f = "g"
+      if ((flags1 & CASE_INSENSITIVE) != 0)
+        f += "i"
+      if ((flags1 & MULTILINE) != 0)
+        f += "m"
+      f
```

Comment:
Could also create a function that tests if the flag is on. Something that would be used like `containsFlag(flags1, CASE_INSENSITIVE)`. I see at least 3 places where it would be used.